Shared/Samples/Navigate a route/NavigateARouteView.swift

```diff
@@ -0,0 +1,282 @@
+                routeGeometry,
+                maxSegmentLength: 50,
+                lengthUnit: .meters,
+                curveType: .geodesic
+            ) as! Polyline
+            let mockDataSource = SimulatedLocationDataSource(polyline: densifiedRoute)
+            
+            // Creates a route tracker location data source.
+            let routeTrackerLocationDataSource = RouteTrackerLocationDataSource(
+                routeTracker: routeTracker,
+                locationDataSource: mockDataSource
+            )
+            
+            // Sets the location display's data source.
+            locationDisplay.dataSource = routeTrackerLocationDataSource
+            
+            // Updates the graphics and viewpoint.
+            routeAheadGraphic.geometry = routeGeometry
+            viewpoint = Viewpoint(targetExtent: routeGeometry, rotation: 0)
+        }
+        
+        func startUpdates() async {
+            await withTaskGroup(of: Void.self) { group in
+                group.addTask {
+                    await self.updateAutoPanMode()
+                }
+            }
+        }
+        
+        /// Updates the current auto-pan mode if it does not match the location display's auto-pan mode.
+        private func updateAutoPanMode() async {
+            for await mode in locationDisplay.$autoPanMode {
+                if autoPanMode != mode {
+                    autoPanMode = mode
+                }
+            }
+        }
+        
+        /// Starts navigating the route.
+        func startNavigation() async {
+            print(locationDisplay.dataSource.status)
+            do {
+                try await locationDisplay.dataSource.start()
+                locationDisplay.autoPanMode = .navigation
+            } catch {
+                self.error = error
+                isShowingAlert = true
+            }
+        }
+        
+        func resetNavigation() async {
+            print(locationDisplay.dataSource.status)
+            locationDisplay.autoPanMode = .off
+            await locationDisplay.dataSource.stop()
+            setNavigation()
+        }
+    }
+}
+
+private extension Stop {
+    /// The stops for this sample.
+    static var stops: [Stop] {
+        let one = Stop(point: Point(x: -117.160386727, y: 32.706608, spatialReference: .wgs84))
+        one.name = "San Diego Convention Center"
+        let two = Stop(point: Point(x: -117.173034, y: 32.712329, spatialReference: .wgs84))
+        two.name = "USS San Diego Memorial"
+        let three = Stop(point: Point(x: -117.147230, y: 32.730467, spatialReference: .wgs84))
+        three.name = "RH Fleet Aerospace Museum"
+        return [one, two, three]
+    }
+}
+
+private extension URL {
+    /// The URL for the route task.
+    static var routeTask: URL {
+        URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/Route")!
```

Comment:
This needs to be updated, see the discuss from Shubham in the samples team channel.